### PR TITLE
Add point model binder.

### PR DESCRIPTION
**Pull request by KonovalovaVV**

### CODE REVIEW

File: CustomModelBinder/CustomModelBinder/Models/Point.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public class Point
+    {
+        public int x { get; set; }
+        public int y { get; set; }
+        public int z { get; set; }
+        
+        public static Point Parse(string input)
+        {
+            string[] values = input.Split(',');
```
Comment (MaksHladki) [inline CustomModelBinder/CustomModelBinder/Models/Point.cs:15]: Please, refactor the code.
Example:
image

File: CustomModelBinder/CustomModelBinder/Points.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"x":44,"y":23,"z":65}{"x":1,"y":2,"z":3}{"x":1,"y":2,"z":3}
```
Comment (MaksHladki) [inline CustomModelBinder/CustomModelBinder/Points.json:1]: Please, remove this file.

File: CustomModelBinder/CustomModelBinder/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddControllersWithViews(opts =>
+            {
+                opts.ModelBinderProviders.Insert(0, new CustomPointModelBinderProvider());
```
Comment (MaksHladki) [inline CustomModelBinder/CustomModelBinder/Startup.cs:23]: Try to use the ModelBinder attribute instead.
``` [ModelBinder(BinderType = typeof(PointViewModelBinder))]```

File: CustomModelBinder/CustomModelBinder/Controllers/HomeController.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+using Microsoft.Extensions.Logging;
+using CustomModelBinder.Models;
+using System.Collections.Generic;
+
+namespace CustomModelBinder.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly ILogger<HomeController> _logger;
+        private static List<Point> points = new List<Point>();
```
[the remark this answers is not in the document]
Comment (KonovalovaVV) [reply CustomModelBinder/CustomModelBinder/Controllers/HomeController.cs:12]: Ok, remove it

File: CustomModelBinder/CustomModelBinder/Controllers/HomeController.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
+﻿using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using CustomModelBinder.Models;
+using System.Collections.Generic;
+
+namespace CustomModelBinder.Controllers
+{
+    public class HomeController : Controller
```
[the remark this answers is not in the document]
Comment (KonovalovaVV) [reply CustomModelBinder/CustomModelBinder/Controllers/HomeController.cs:9]: Ok, in this branch I add PointController

File: CustomModelBinder/CustomModelBinder/Controllers/HomeController.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+        [HttpPost]
+        public IActionResult Create(Point p)
+        {
+            points.Add(p);
+            p.Serialize();      
+            return RedirectToAction("Index");
+        }
+
+        public IActionResult Privacy()
```
[the remark this answers is not in the document]
Comment (KonovalovaVV) [reply CustomModelBinder/CustomModelBinder/Controllers/HomeController.cs:37]: Ok

File: CustomModelBinder/CustomModelBinder/Infrastructure/CustomPointBinder.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
+﻿using CustomModelBinder.Models;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System;
+using System.Threading.Tasks;
+
+namespace CustomModelBinder.Infrastructure
+{
+    public class CustomPointBinder : IModelBinder
```
[the remark this answers is not in the document]
Comment (KonovalovaVV) [reply CustomModelBinder/CustomModelBinder/Infrastructure/CustomPointBinder.cs:8]: Ok.

File: CustomModelBinder/CustomModelBinder/Models/Point.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+using System.Text.Json;
+
+namespace CustomModelBinder.Models
+{
+    public class Point
+    {
+        public int x { get; set; }
```
[the remark this answers is not in the document]
Comment (KonovalovaVV) [reply CustomModelBinder/CustomModelBinder/Models/Point.cs:9]: Ok

File: CustomModelBinder/CustomModelBinder/Models/Point.cs (hunk 1 above)
Comment (KonovalovaVV) [reply CustomModelBinder/CustomModelBinder/Models/Point.cs:15]: Ok

File: CustomModelBinder/CustomModelBinder/Points.json (hunk 2 above)
Comment (KonovalovaVV) [reply CustomModelBinder/CustomModelBinder/Points.json:1]: Ok

**Pull request merged**